bufferevent_sock.c

```diff
@@ -373,9 +373,9 @@ bufferevent_socket_connect(struct bufferevent *bev,
 		fd = socket(sa->sa_family, SOCK_STREAM, 0);
 		if (fd < 0)
 			goto done;
-		if (evutil_make_socket_nonblocking(fd)<0)
-			goto done;
 		ownfd = 1;
+		if (evutil_make_socket_nonblocking(fd)<0)
+			goto freesock;
```

Comment:
Nice catch, but please clean this patch and change subject/description of the patch/pull-request to describe this change, not other bits (that eventually removed in #600)